functions-python/feed_sync_process_transitland/src/main.py

```diff
@@ -0,0 +1,381 @@
+                    f"Updated external ID mapping to new feed_id: {new_feed_id}"
+                )
+
+            # Create redirect
+            redirect = Redirectingid(source_id=old_feed_id, target_id=new_feed_id)
+            self.session.add(redirect)
+            logger.debug(f"Created redirect from {old_feed_id} to {new_feed_id}")
+
+            # Flush changes to get IDs
+            self.session.flush()
+
+            logger.info(
+                f"Updated feed for external_id: {payload.external_id}, new feed_id: {new_feed_id}"
+            )
+
+        except Exception as e:
+            logger.error(
+                f"Error updating feed for external_id {payload.external_id}: {str(e)}"
+            )
+            raise
+
+    def check_feed_url_exists(self, feed_url: str) -> bool:
+        """
+        Check if a feed with the given URL already exists
+
+        Args:
+            feed_url (str): The URL to check
+
+        Returns:
+            bool: True if URL exists, False otherwise
+        """
+        result = (
+            self.session.query(Feed)
+            .filter(Feed.producer_url == feed_url, Feed.status == "active")
+            .first()
+        )
+
+        if result is not None:
+            logger.debug(f"Found existing feed with URL: {feed_url}")
+            return True
+
+        logger.debug(f"No existing feed found with URL: {feed_url}")
+        return False
+
+    def get_current_feed_info(
+        self, external_id: str, source: str
+    ) -> Tuple[Optional[str], Optional[str]]:
+        """
+        Get current feed ID and URL for given external ID
+
+        Args:
+            external_id (str): The external ID to look up
+            source (str): The source of the feed
+
+        Returns:
+            Tuple[Optional[str], Optional[str]]: Tuple of (feed_id, feed_url)
+        """
+        result = (
+            self.session.query(Feed.id, Feed.producer_url)
+            .join(Externalid)
+            .filter(
+                Externalid.associated_id == external_id,
+                Externalid.source == source,
+                Feed.status == "active",
+            )
+            .first()
+        )
+        if result:
+            logger.debug(
+                f"Retrieved current feed " f"info for external_id: {external_id}"
+            )
+            return result[0], result[1]
+
+        logger.debug(f"No existing feed found for external_id: {external_id}")
+        return None, None
+
+    def publish_to_batch_topic(self, payload: FeedPayload) -> None:
+        """
+        Publish feed to dataset batch topic
+
+        Args:
+            payload (FeedPayload): The feed payload to publish
+        """
+        topic_path = self.publisher.topic_path(PROJECT_ID, DATASET_BATCH_TOPIC)
+        logger.debug(f"Publishing to topic: {topic_path}")
+
+        data = json.dumps(
+            {"feed_id": payload.feed_id, "execution_id": payload.execution_id}
+        ).encode("utf-8")
```

Comment:
This is not the message format that the dataset processing topic expects. It should be:
```json
{
        "message": {
            "data": 
            {
                "execution_id":  "execution_id",
                "producer_url":  "producer_url",
                "feed_stable_id":  "feed_stable_id",
                "feed_id":  "feed_id",
                "dataset_id":  "dataset_id",
                "dataset_hash":  "dataset_hash",
                "authentication_type":  "authentication_type",
                "authentication_info_url":  "authentication_info_url",
                "api_key_parameter_name": "api_key_parameter_name"
            }            
        }
    }
```
Refer to `functions-python/batch_process_dataset` for more information.